### PR TITLE
comment danger of potential future VM.version increases

### PR DESCRIPTION
**Pull request by tbjump**
This change is [Reviewable](https://reviewable.io/reviews/certusone/wormhole/1401)

### CODE REVIEW

**Comment by kcsongor** [comment]
In this case the choice to not include the version number is by design, the message digest is only concerned with the observation itself (which integrators can safely use to implement replay protection). This allows emitting newer types of VAAs in a backwards compatible fashion - the core contract just has to know how to derive the message digest based on the version number.

**Comment by kcsongor** [comment]
I'll close this, but feel free to reopen if you think further discussion is warranted!

**Comment by tbjump** [comment]
That's a good insight, which I would like to have commented in the code. I update the comments accordingly. Please re-review.

File: ethereum/contracts/Messages.sol [hunk 1]
```diff
@@ -127,7 +125,13 @@ contract Messages is Getters {
             index += 1;
         }
 
-        // Hash the body
+        /*
+        Hash the body
+
+        SECURITY: Do not change the way the hash of a VM is computed! 
+        Changing it could result into two different hashes for the same observation. 
+        But xDapps rely on the hash of an observation for reply protection.
```
Comment (kcsongor) [inline ethereum/contracts/Messages.sol:142]: replay

File: node/pkg/vaa/structs.go [hunk 2]
```diff
@@ -429,6 +429,10 @@ func (v *VAA) HexDigest() string {
 	return hex.EncodeToString(v.SigningMsg().Bytes())
 }
 
+/*
+SECURITY: Do not change this code! Changing it could result into two different hashes for
```
Comment (kcsongor) [inline node/pkg/vaa/structs.go:439]: result in

File: node/pkg/vaa/structs.go [hunk 3]
```diff
@@ -429,6 +429,10 @@ func (v *VAA) HexDigest() string {
...
+the same observation. But xDapps rely on the hash of an observation for reply protection.
```
Comment (kcsongor) [inline node/pkg/vaa/structs.go:440]: replay

**Review by kcsongor — APPROVED** [review]

**Review by claudijd — APPROVED** [review]

**Pull request merged**